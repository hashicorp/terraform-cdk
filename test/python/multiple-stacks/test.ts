// Copyright (c) HashiCorp, Inc
// SPDX-License-Identifier: MPL-2.0
import { TestDriver } from "../../test-helper";

describe("python full integration test synth", () => {
  let driver: TestDriver;

  beforeAll(async () => {
    driver = new TestDriver(__dirname);
    await driver.setupPythonProject();
  });

  test("synth generates JSON for both stacks", async () => {
    await driver.synth();
    expect(
      driver.synthesizedStack("python-simple-one").toString()
    ).toMatchSnapshot();
    expect(
      driver.synthesizedStack("python-simple-two").toString()
    ).toMatchSnapshot();
  });
});
