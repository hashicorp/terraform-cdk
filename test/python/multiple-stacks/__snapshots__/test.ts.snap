// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`python full integration test synth synth generates JSON for both stacks 1`] = `
"{
  "//": {
    "metadata": {
      "backend": "local",
      "stackName": "python-simple-one",
      "version": "stubbed"
    },
    "outputs": {}
  },
  "provider": {
    "null": [
      {}
    ]
  },
  "resource": {
    "null_resource": {
      "null-resource": {
        "//": {
          "metadata": {
            "path": "python-simple-one/null-resource",
            "uniqueId": "null-resource"
          }
        }
      }
    }
  },
  "terraform": {
    "backend": {
      "local": {
        "path": "terraform.tfstate"
      }
    },
    "required_providers": {
      "null": {
        "source": "hashicorp/null",
        "version": "3.1.1"
      }
    }
  }
}"
`;

exports[`python full integration test synth synth generates JSON for both stacks 2`] = `
"{
  "//": {
    "metadata": {
      "backend": "local",
      "stackName": "python-simple-two",
      "version": "stubbed"
    },
    "outputs": {}
  },
  "provider": {
    "null": [
      {}
    ]
  },
  "resource": {
    "null_resource": {
      "null-resource": {
        "//": {
          "metadata": {
            "path": "python-simple-two/null-resource",
            "uniqueId": "null-resource"
          }
        }
      }
    }
  },
  "terraform": {
    "backend": {
      "local": {
        "path": "terraform.tfstate"
      }
    },
    "required_providers": {
      "null": {
        "source": "hashicorp/null",
        "version": "3.1.1"
      }
    }
  }
}"
`;
